lxd/storage/backend_lxd.go

```diff
@@ -1645,9 +1667,27 @@ func (b *lxdBackend) CreateInstanceFromImage(inst instance.Instance, fingerprint
 	revert := revert.New()
 	defer revert.Fail()
 
+	volumeConfig := make(map[string]string)
+	err = b.applyInstanceRootDiskInitialValues(inst, volumeConfig)
+	if err != nil {
+		return err
+	}
+
+	// If the volume config is not empty after applying the initial configuration, it means that initial
+	// configuration values are present in the volume config. In such case, we should use an optimized
+	// image only if the volume configuration matches the pool's default settings for new volumes.
+	// Otherwise, the image should be directly unpacked into the instance's volume.
+	useOptimizedImage := true
```

Comment:
Lets do this here:

```go
useOptimizedImage := b.driver.Info().OptimizedImages 
```

Then we only need to invoke b.shouldUseOptimizedImage if useOptimizedImage is true.

Then on line 1709 we can then just use :

```go
if !useOptimizedImage  {}
```

Reply:
Thar way it keeps the logic together rather than spreading it throughout the function.